server/user_server.go

```diff
@@ -41,13 +37,24 @@ func httpOAuth1Complete(jsi *jiraServerInstance, w http.ResponseWriter, r *http.
 			errors.WithMessage(appErr, "failed to load user "+mattermostUserId)
 	}
 
+	oauthTmpCredentials, err := jsi.Plugin.OneTimeLoadOauth1aTemporaryCredentials(mattermostUserId)
+	if err != nil || oauthTmpCredentials == nil || len(oauthTmpCredentials.Token) <= 0 {
+		return http.StatusInternalServerError, errors.WithMessage(err, "failed to get temporary credentials for "+mattermostUserId)
+	}
+
+	if oauthTmpCredentials.Token != requestToken {
```

Comment:
This check prevents reuse (since the credentials are deleted) and also makes sure the MM user that started the flow is the one finishing it (keyed on userid)